src/lib.rs

```diff
@@ -123,15 +134,26 @@ impl Contract {
         self.posts.push(&post.into());
         self.post_to_parent.insert(&id, &parent_id);
 
-        let mut siblings = self
-            .post_to_children
-            .get(&parent_id)
-            .unwrap_or_else(|| panic!("Parent id {} not found", parent_id));
+        let mut siblings = self.post_to_children.get(&parent_id).unwrap_or_else(
```

Comment:
If someone passes an existing `parent_id` it should always be the case that `post_to_children` contains the record for it, unless `parent_id==ROOT_POST_ID`. What is the use of this change?

Reply:
If the intent to make it work with `parent_id ==ROOT_POST_ID` then we can just do an early return of `None` here.

Reply:
I'll revert this line, let's address this problem in a separate PR.

> If the intent to make it work with parent_id ==ROOT_POST_ID 

Yes this was intend to bootstrap the state - where I first deploy contract and create first post it panics here because initially root id doesn't have siblings (`post_to_children[ROOT_ID]` is `None`).

> then we can just do an early return of None here.

Do you mean something like:
```
if parent_id != ROOT_POST_ID {
...
} else {
// doesn't do anything with top level post's sibling, aka ROOT_POST's children.
  return None
}
 ```

But in `main` branch, after insert an top-level post, this line:
```
        self.post_to_children.insert(&parent_id, &siblings);
```
will insert/update an entry `ROOT_ID` -> `vector of all top level posts`. Do we want to keep this entry?

Reply:
The current code of this PR looks good.